### DISCUSSION TITLE
Datasource MySQL / MariaDB & Multiuser support

### DISCUSSION THREAD

**Discussion by promexio** [frappe/books#650] [Ideas]
On DB selection, an option shows up to connect to a SQL instance (on prem or in the cloud). 
Client will have some table locking functionality in order to avoid multisession issues. 

What do you mean? 
Would that serve anybody? 
How big would be the effort behind it? 

I think it could be a huge benefit for accounting firms - even small companies or holding coorporations could replace their ERP systems often too complex and too heavy for the applied use cases.

**Comment by promexio** [comment]
-- Any news?

**Comment by promexio** [comment]
-- Any news?

**Comment by Isaac-GC** [comment]
Hi @promexio, 

There have been similar asks for this and is on the roadmap, but overall will be a large undertaking and will take some time to flush out.

Hopefully the bugs get worked out and work on other minor features get pushed (Dark Mode, Being able to print using the system printer,  etc). Once those are pushed, I think the next move is to decouple as much as possible from electron's desktop pieces, which will also involve adding various database types support.
